client/vmmCtrlrP_service.go

```diff
@@ -0,0 +1,259 @@
+	dn := fmt.Sprintf("uni/vmmp-%s/dom-%s/ctrlr-%s", provider_profile_vendor, vmm_domain, name)
+	return sm.DeleteByDn(dn, models.VmmctrlrpClassName)
+}
+
+func (sm *ServiceManager) UpdateVMMController(name string, vmm_domain string, provider_profile_vendor string, description string, vmmCtrlrPattr models.VMMControllerAttributes) (*models.VMMController, error) {
+	rn := fmt.Sprintf("ctrlr-%s", name)
+	parentDn := fmt.Sprintf("uni/vmmp-%s/dom-%s", provider_profile_vendor, vmm_domain)
+	vmmCtrlrP := models.NewVMMController(rn, parentDn, description, vmmCtrlrPattr)
+
+	vmmCtrlrP.Status = "modified"
+	err := sm.Save(vmmCtrlrP)
+	return vmmCtrlrP, err
+
+}
+
+func (sm *ServiceManager) ListVMMController(vmm_domain string, provider_profile_vendor string) ([]*models.VMMController, error) {
+
+	baseurlStr := "/api/node/class"
+	dnUrl := fmt.Sprintf("%s/uni/vmmp-%s/dom-%s/vmmCtrlrP.json", baseurlStr, provider_profile_vendor, vmm_domain)
+
+	cont, err := sm.GetViaURL(dnUrl)
+	list := models.VMMControllerListFromContainer(cont)
+
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsMcastAddrNs(parentDn, tnFvnsMcastAddrInstPName string) error {
+	dn := fmt.Sprintf("%s/rsmcastAddrNs", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tnFvnsMcastAddrInstPName": "%s"
+			}
+		}
+	}`, "vmmRsMcastAddrNs", dn, tnFvnsMcastAddrInstPName))
+
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationvmmRsMcastAddrNs(parentDn string) error {
+	dn := fmt.Sprintf("%s/rsmcastAddrNs", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsMcastAddrNs")
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsAcc(parentDn, vmmUsrAccPDn string) error {
+	dn := fmt.Sprintf("%s/rsacc", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
+			}
+		}
+	}`, "vmmRsAcc", dn, vmmUsrAccPDn))
+
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationvmmRsAcc(parentDn string) error {
+	dn := fmt.Sprintf("%s/rsacc", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsAcc")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVmmCtrlrP(parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rsvmmCtrlrP-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.